framework/src/utils/PointReduction.C

```diff
@@ -0,0 +1,98 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PointReduction.h"
+#include "MooseError.h"
+
+#include <algorithm>
+#include <cmath>
+
+namespace PointReduction
+{
+
+Real
+sqr(Real a)
+{
+  return a * a;
+}
+
+Real
+perpendicularDistance(const FunctionNode & node,
+                      const FunctionNode & begin,
+                      const FunctionNode & end)
+{
+  const Real x0 = node.first;
+  const Real y0 = node.second;
+  const Real x1 = begin.first;
+  const Real y1 = begin.second;
+  const Real x2 = end.first;
+  const Real y2 = end.second;
+
+  const Real denom = std::sqrt(sqr(y2 - y1) + sqr(x2 - x1));
+  mooseAssert(denom > 0, "Line begin and end points bust not be the same");
```

Comment:
bust?

Reply:
This is probably a good place to use MooseUtils::absoluteFuzzyGreaterThan() so you don't divide by a really small number either.